레이아웃 및 목록 페이지, 상세 페이지 구현

NiNinanana opened the pull request:
## 수정
- 메인 페이지 레이아웃 구현
- 헤더, 푸터, 채용 공고 섹션 구현
- global.css는 모든 페이지에 필요한 css를 담음
- 목록 페이지, 상세 페이지를 javascript로 구현

NiNinanana committed:
메인 페이지 레이아웃 구현

NiNinanana committed:
목록 페이지, 상세 페이지 javascript로 구현
각 컴포넌트에 대하여 css 파일 구성
layout 컴포넌트 만들어 고차함수로 구현

BuildTheTruth commented on utils/css.js:
```diff
@@ -0,0 +1,9 @@
+const applyStyle = (path) => {
+  const linkElement = document.createElement("link");
```
컴포넌트마다 style link가 추가되는건 좋은 방법은 아닌 것 같습니다. 그 이유에 대해 검토해보고 vanillaJS에서 css를 주로 어떻게 사용하는지 구글링해보는 것을 추천드립니다.

NiNinanana replied:
하나의 index.css 파일에 import해서 사용하는 방식을 찾아봤어요. 그래서 그렇게 적용을 했습니다.

BuildTheTruth commented on components/layout/index.js:
```diff
@@ -0,0 +1,101 @@
+          </address>
+          <ul class="social-list">
+            <li>
+              <button><img src="/assets/icons/facebook.svg" /></button>
```
반복되는 부분은 map를 사용해보는 것도 좋을 것 같네요.

BuildTheTruth commented on css/vote.css:
```diff
@@ -0,0 +1,57 @@
+  margin: 0px 12px;
+}
+
+#share-button {
```
혹시 id를 사용하신 이유가 있을까요?

NiNinanana replied:
코드의 딱 한 곳에서만 사용하기 위해서 id를 사용해서 스타일을 적용했습니다

BuildTheTruth commented on components/layout/page.js:
```diff
@@ -0,0 +1,18 @@
+import { applyStyle } from "../../utils/css.js";
+
+const Page = (content, props) => {
```
content와 props를 분리하신 이유가 있을까요?

NiNinanana replied:
content는 componentDidMount와 render만 들어가면 좋을까 생각도 했는데 그냥 content 객체 안에 다 넣어도 되겠다는 생각을 했습니다. 그래서 content안에 props 프로퍼티를 만들어 수정했습니다.

NiNinanana committed:
코드리뷰 받아 수정
css를 index.css만 link가 만들어지도록 하고 나머지 css는 index.css에 import 되도록 설정
반복되는 부분에 map 사용
content와 props를 분리하지 않고 content 안에서 다 사용하도록 수정

NiNinanana committed:
css를 전부 class로 변경, content 대신 props 사용

NiNinanana commented:
@f-lab-dennis 이 PR은 여기서 마무리하고 스타일에 관한 부분은 다른 PR에서 진행하도록 하겠습니다.
